lib/adb-device.js

```diff
@@ -190,6 +190,64 @@ class ADBDevice {
     return data;
   }
 
+  // Receive a file listing and parse it into an array of file objects.
+  async recvList(localId, remoteId) {
+    // This is the structure of each directory entry returned by LIST
+    // see https://android.googlesource.com/platform/system/core/+/master/adb/SYNC.TXT#39
+    const props = [
+      'cmd',
+      'mode',
+      'size',
+      'modified',
+      'length',
+      'filename'
+    ];
+    const files = [];
+    let file = {};
+    let i = 0;
+
+    // Keep requesting new packets until we receive 'DONE'
+    // TODO: Do we need a timeout in case 'DONE' is never sent? Shouldn't happen
+    while (true) {
+      let packet = await this.recvAndOkay(localId, remoteId);
+      let pos = 0;
+      do {
```

Comment:
The problem is that in testing these pieces are split across different packets, or combined. I.e. you might get back:

- packet 1: `DENT`, `mode`,`size`,`modified`,`length`
- packet 2: `filename`,`DENT`,`mode`,`size`,`modified`,`length`,`filename`,`DENT`,`mode`
- packet 3: `size`,`modified`,`length`,`filename`,`DONE`, (then sometimes a random number of zeros)